src/libra_py/dynamics/qtag/QTAG_basis.py

```diff
@@ -0,0 +1,119 @@
+"""
+..module:: QTAG_basis
+  :platform: Unix, Windows
+  :synopsis: This module contains update functions for the {q,p,a,s} basis components.
+
+..moduleauthors :: Matthew Dutra
+"""
+
+from liblibra_core import *
+from QTAG_config import univ
+
+def frozen(i,param_in,*args):
+	"""The non-update function.
+        Args:
+		i (integer) : The DoF being considered
+
+                param_in (MATRIX): 1-by-ntraj basis parameter matrix for q, p, a, or s.
+
+	Returns:
+		param_in (MATRIX): The same input parameter matrix q, p, a, or s.
+        """
+
+	return(param_in)
+
+def q_update(ndim,q_in,mom_in):
+	"""Returns the updated position *q_out* from an input position *q_in* and the momentum *mom*, calculated via q_out = q_in+mom*dt/m (i.e. symplectic).
+
+        Args:
+		ndim (integer): The dimension along which q is being updated.
+
+                q_in (MATRIX): The 1-by-ntraj basis position MATRIX q to be updated.
+
+		mom_in (MATRIX): The 1-by-ntraj momentum MATRIX p corresponding to q.
+        Returns:
+                q_out (MATRIX): Updated 1-by-ntraj basis position MATRIX q.
+        """
+
+	ntraj=univ['ntraj']
+	q_out=MATRIX(1,ntraj)
```

Comment:
in Python, you don't have to initialize the results, also combining the with what i see next, you could do just 
`return q_in + mom_in * univ['dt']/univ['mass'][ndim]`